packages/runtime/test/lib/testUtils.ts

```diff
@@ -417,13 +417,38 @@ export async function establishPendingRelationshipWithRequestFlow(
 
 export async function ensureActiveRelationship(sTransportServices: TransportServices, rTransportServices: TransportServices): Promise<RelationshipDTO> {
     const rTransportServicesAddress = (await rTransportServices.account.getIdentityInfo()).value.address;
-    const relationships = (await sTransportServices.relationships.getRelationships({ query: { peer: rTransportServicesAddress } })).value;
-    if (relationships.length === 0) {
+    const sRelationships = (await sTransportServices.relationships.getRelationships({ query: { peer: rTransportServicesAddress } })).value;
+
+    const sTransportServicesAddress = (await sTransportServices.account.getIdentityInfo()).value.address;
+    const rRelationships = (await rTransportServices.relationships.getRelationships({ query: { peer: sTransportServicesAddress } })).value;
+
+    if (sRelationships.length === 0 && rRelationships.length === 0) {
         await establishRelationship(sTransportServices, rTransportServices);
-    } else if (relationships[0].status === RelationshipStatus.Pending) {
-        const relationship = relationships[0];
-        await sTransportServices.relationships.acceptRelationship({ relationshipId: relationship.id });
+    } else if (sRelationships.length === 0 && rRelationships.length !== 0) {
```

Comment:
Could we (currently) run into a problem here, since the recipient might have a rejected or revoked Relationship and not like you implicitly assume a deletion proposed Relationship? Maybe add a check for safety (and even if revoked and rejected Relationships are directly deleted in the future) and for better understanding.

Reply:
Same thing if the sender has a rejected or revoked Relationship.